pkg/middlewares/auth/forward.go

```diff
@@ -195,7 +217,41 @@ func (fa *forwardAuth) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	req.RequestURI = req.URL.RequestURI()
-	fa.next.ServeHTTP(rw, req)
+	fa.next.ServeHTTP(headers.NewResponseModifier(rw, req, cac.PostRequestModifyResponseHeaders), req)
```

Comment:
This removes the need to check the length in `PostRequestModifyResponseHeaders` entirely. As `PostRequestModifyResponseHeaders` can't panic due to a zero length list and should never reach this point due to this change we can simplify the code a lot.

```suggestion
	switch len(cac.authResponseCookieHeaders) {
	case 0:
		fa.next.ServeHTTP(rw, req)
	default:
		fa.next.ServeHTTP(headers.NewResponseModifier(rw, req, cac.PostRequestModifyResponseHeaders), req)
	}
```